auth_backend/backends.py

```diff
@@ -34,10 +34,25 @@ def authenticate(self, email=None, username=None, password=None, **kwargs):
         if strategy:
             payload['strategy'] = strategy
 
+        auth_api_client = AuthApiClient(cas_credentials)
         status, data = auth_api_client.call('sessions', 'POST', payload)
 
         if status not in (200, 404,):
             raise CASUnexpectedStatusCode(status, data)
 
         if status == 200:
-            return user
+            if existing_user:
+                existing_user.cas_credentials = cas_credentials
+            else:
+                try:
```

Comment:
i need the finally block, if there is an exception the client (as in say the CMS) must catch the exception and do something like 'sorry we cant log you in right now', but we still need to clean up here